stan/math/prim/scal/meta/enable_if_arithmetic.hpp

```diff
@@ -0,0 +1,12 @@
+#ifndef STAN_MATH_PRIM_SCAL_META_ENABLE_IF_ARITHMETIC_HPP
+#define STAN_MATH_PRIM_SCAL_META_ENABLE_IF_ARITHMETIC_HPP
+
+#include <type_traits>
+
+namespace stan {
+
+template <typename T>
+using enable_if_arithmetic = std::enable_if_t<std::is_arithmetic<T>::value>;
```

Comment:
Great! Doing this metaprogram was the first thing that I thought of when I started looking at the changes in this PR yesterday.

Reply:
+1 on this metaprogram. Would it workable to have a variadic version, like ```enable_if_all_arithmetic``` (or something less wordy)? That could help condense some of the templating from:
```
template <typename T1, typename T2, typename = enable_if_arithmetic<T1>,
                  typename = enable_if_arithmetic<T2>>
```
to
```
template <typename T1, typename T2, typename = enable_if_all_arithmetic<T1, T2>>
```

Reply:
Yep! We can use `conjuction` and `disjunction` for `enable_if_all_<blah>` and `enable_if_any_<blah>`